### DISCUSSION TITLE
How to use several different DataTemplates in TabControl.ContentTemplate?

### DISCUSSION THREAD

**Discussion by kochkaqqq** [AvaloniaUI/Avalonia#14474] [Q&A] [answered]
< TabControl.ContentTemplate>
    < DataTemplate DataType="pgs:Type1">
        < TextBlock Text="{Binding Type1Property}" />
    </ DataTemplate>

    < DataTemplate DataType="pgs:Type2">
        < TextBlock Text="{Binding Type2Property}" />
    </ DataTemplate>
</ TabControl.ContentTemplate>

Any ways to make it work?

PS Type1 and Type2 inherit from same class.

**Comment by timunie** [comment, marked as answer]
Several options. I'd implement a FuncDataTemplate or IDataTemplate. Check Avalonia.Samples
